combine/combine_test.go

```diff
@@ -0,0 +1,293 @@
+package combine
+
+import (
+	"bufio"
+	"compress/gzip"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestCombine(t *testing.T) { //nolint:gocognit
+	Convey("The function combines the files", t, func() {
```

Comment:
There should a TestXXX function and explanatory Convey message for each general public function. The tests should confirm the function does what it's supposed to do.

You might have to walk me through what you have here with me, but I mostly just see err checks and the file-type specific tests that we don't need here for now. I also don't understand how those file-type specific tests work.